### PR TITLE
DDF-1174: Added JaCoCo to plug-in management section

### PR DESCRIPTION
**Pull request by lessarderic**
Added default JaCoCo configuration to plug-in management section.

[Review on Reviewable](https://reviewable.io/reviews/codice/ddf-parent/7)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -397,6 +397,63 @@
                         </additionalparam>
                     </configuration>
                 </plugin>
+
+                <plugin>
+                    <groupId>org.jacoco</groupId>
+                    <artifactId>jacoco-maven-plugin</artifactId>
+                    <version>0.7.2.201409121644</version>
```
Comment (mrmateo) [inline pom.xml]: Looks like there is a newer version of this plugin - 0.7.4.201502262128. Not sure if it has any changes that would apply to DDF, but it is about 5 months newer than 0.7.2.

**Comment by shaundmorris** [comment]
<sup>**[pom.xml, line 404 [r2]](https://reviewable.io:443/reviews/codice/ddf-parent/7#-JnwlWrUYzdd9x0k7y_C)** ([raw file](https://github.com/codice/ddf-parent/blob/4909012c31a0c4778f48519f5a8579d3f38ffed7/pom.xml#L404)):</sup>
I am not sure about the latest either but generally we like to upgrade while we can so we don't have stale/unsupported versions out there. 

---

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-parent/7)

**Comment by shaundmorris** [comment]
Any other thoughts on JaCoco? Since there is a domino affect here on pull requests I would like to see if we can come to quick resolution on merging the JaCoco related PRs. 

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-parent/7)

**Comment by mrmateo** [comment]
Yeah, I had to look up what that minimum limit setting meant. So it looks like if the code coverage doesn't meet those minimums then the build will fail? 

So all the individual settings are to specify at what that minimum is for each project?

**Comment by lessarderic** [comment]
That's correct.

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-parent/7)

**Comment by shaundmorris** [comment]
@mrmateo any other comments/questions? Good questions already asked, just to wanted to make sure they were all answered. Overall this sets up DDF to actually enforce code coverage at a certain point if decided that is the route to go.

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-parent/7)

**Comment by mrmateo** [comment]
I'm good with merging the parent. It looks like some of the other PRs still duplicate what the parent is setting. I did a quick look and thought I saw about half of the files in catalog duplicate the main parent.

**Comment by shaundmorris** [comment]
@mrmateo +1. We need to verify we aren't creating extra maintenance in the other PRs. That will need to be cleaned-up prior to approving the other PRs. 

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-parent/7)

**Comment by shaundmorris** [comment]
Reviewed files:
- pom.xml @ r2

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-parent/7)

**Comment by mrmateo** [comment]
@shaundmorris - from your wording it sounds like the code coverage won't be enforced after this PRs? Just want to make sure I understand what will happen once this is all merged to prevent issues like when we added checkstyle and it caused some issues with other developers trying to merge their code.

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-parent/7)

**Comment by lessarderic** [comment]
After this PR has been merged, the build will fail if the code coverage goes below the thresholds that were set. Default is 75% for those that don't have anything set explicitly. Our goal is to eventually increase those numbers over time up to 75% across the board so we can get decent code coverage on the code base. This first step is basically to make sure we're not moving backwards :-)

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-parent/7)

**Comment by shaundmorris** [comment]
@mrmateo - correct. +1 @lessarderic. So I meant that we were not enforcing an exact coverage percentage as a means of compliance. However, with these PRs we can at least view any gaps and low percentage of code coverage. 

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-parent/7)

**Comment by mrmateo** [comment]
Gotcha, OK. Thanks for the clarifications. That makes perfect sense. 

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-parent/7)

**Comment by shaundmorris** [comment]
We have a build issue that needs to be corrected first. "Error: Could not find or load main class ${argLine}"

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-parent/7)

**Comment by shaundmorris** [comment]
Okay I did a full build using this branch and the build was fine. So we should be good to go here now.

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-parent/7)

**Pull request merged**